activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -0,0 +1,31 @@
+module ActiveRecord
+  module CollectionCacheKey
+
+    # Generates a cache key for the records in the given collection.
+    # See <tt>ActiveRecord::Relation#cache_key</tt> for details.
+    def collection_cache_key(collection = all, timestamp_column = :updated_at)
```

Comment:
I would actually prefer to mark this method as `# :nodoc:` since it only serves as a hook for `Relation#cache_key`